pkg/azure/client/storage.go

```diff
@@ -28,9 +30,46 @@ type BlobStorageClient struct {
 	client *azblob.Client
 }
 
-// newStorageClient creates a client for an Azure Blob storage by reading auth information from secret reference. Requires passing the storage domain (formerly
+// BlobStorageDomainFromCloudConfiguration returns the storage service domain given a known cloudConfiguration.
+func BlobStorageDomainFromCloudConfiguration(cloudConfiguration *azureapi.CloudConfiguration) (string, error) {
+	// Unfortunately the valid values for storage domains run by Microsoft do not seem to be part of any sdk module. They might be queryable from the cloud configuration,
+	// but I also haven't been able to find a documented list of proper ServiceName values.
+	// Furthermore, it seems there is still no unified way of specifying the cloud instance to connect to as the domain remains part of the storage account URL while
+	// the new options _also_ allow configuring the cloud instance.
+	switch {
+	case cloudConfiguration == nil:
+		return "blob.core.windows.net", nil
```

Comment:
Is it possible to make constants for these? (It is a pity that designers of the SDK do not think of these things)
```go
const (
  BlobStoreGlobalDomain = "blob.core.windows.net"
  BlobStoreChinaDomain = "blob.core.chinacloudapi.cn"
  BlobStoreUSGovDomain = "blob.core.usgovcloudapi.net"
```